scrapy/pipelines/files.py

```diff
@@ -334,6 +334,9 @@ def _warn():
 
         media_guid = hashlib.sha1(to_bytes(url)).hexdigest()  # change to request.url after deprecation
         media_ext = os.path.splitext(url)[1]  # change to request.url after deprecation
+        if not media_ext or not all([char.isalpha() or char.isdigit() for char in media_ext[1:]]):
```

Comment:
That url looks super convenient. I think makes sense to keep it for backwards compatibility; but we should add a comment so we easily identify this case in the future.